doc/Caliptra.md

```diff
@@ -866,65 +866,76 @@ Caliptra firmware is composed of two images: an FMC image and an application fir
 
-*Table 16: Firmware manifest preamble*
+*Table 16: Public Key Descriptor*
+
+| Field | Size (bytes) | Description|
+|-------|--------|------------|
+| Key Descriptor Version | 1 | Version of the Key Descriptor. The value must be 0x1 for Caliptra 2.x |
+| Intent | 1 | 0x1 - Vendor  <br> 0x2 - Owner |
+| Key Type | 1 | 0x1 - ECC  <br> 0x2 - LMS <br> 0x3 - MLDSA |
+| Key Hash Count | 1 | Number of public key hashes (n) |
+| Public Key Hash(s) | 48 * n | List of SHA2-384 public key hashes |
+
+*Table 17: Firmware manifest preamble*
+
 
 Fields are little endian unless described otherwise.
 
 | Field | Size (bytes) | Description|
 |-------|--------|------------|
 | Firmware Manifest Marker | 4 | Magic Number marking the start of the package manifest. The value must be 0x434D414E (‘CMAN’ in ASCII)|
 | Firmware Manifest Size | 4 | Size of the full manifest structure |
-| Vendor ECDSA Public Key 1 | 96 | ECDSA P384 public key used to verify the Firmware Manifest Header Signature. <br> **X-Coordinate:** Public Key X-Coordinate (48 bytes, big endian) <br> **Y-Coordinate:** Public Key Y-Coordinate (48 bytes, big endian) |
-| Vendor ECDSA Public Key 2 | 96 | ECDSA P384 public key used to verify the Firmware Manifest Header Signature. <br> **X-Coordinate:** Public Key X-Coordinate (48 bytes, big endian) <br> **Y-Coordinate:** Public Key Y-Coordinate (48 bytes, big endian) |
-| Vendor ECDSA Public Key 3 | 96 | ECDSA P384 public key used to verify the Firmware Manifest Header Signature. <br> **X-Coordinate:** Public Key X-Coordinate (48 bytes, big endian) <br> **Y-Coordinate:** Public Key Y-Coordinate (48 bytes, big endian) |
-| Vendor ECDSA Public Key 4 | 96 | ECDSA P384 public key used to verify the Firmware Manifest Header Signature. <br> **X-Coordinate:** Public Key X-Coordinate (48 bytes, big endian) <br> **Y-Coordinate:** Public Key Y-Coordinate (48 bytes, big endian) |
-| Vendor LMS Public Key 1 | 48 | LMS public key used to verify the Firmware Manifest Header Signature. <br> **tree_type:** LMS Algorithm Type (4 bytes, big endian) Must equal 12. <br> **otstype:** LM-OTS Algorithm Type (4 bytes, big endian) Must equal 7. <br> **id:**  (16 bytes) <br> **digest:**  (24 bytes) |
-| Vendor LMS Public Key 2 | 48 | LMS public key used to verify the Firmware Manifest Header Signature. <br> **tree_type:** LMS Algorithm Type (4 bytes, big endian) Must equal 12. <br> **otstype:** LM-OTS Algorithm Type (4 bytes, big endian) Must equal 7. <br> **id:**  (16 bytes) <br> **digest:**  (24 bytes) |
-| .<br>.<br>|
-| Vendor LMS Public Key 32 | | |
-| ECDSA Public Key Index Hint | 4 | The hint to ROM to indicate which ECDSA public key it should first use.  |
-| LMS Public Key Index Hint | 4 | The hint to ROM to indicate which LMS public key it should first use.  |
+| Firmware Manifest Type | 4 |  **Byte0:** - Type <br> 0x1 – ECDSA & LMS Keys <br> 0x2 – ECDSA & MLDSA Keys <br> **Byte1-Byte3:** Reserved|
```

Comment:
I know this is what we agreed on, but I was talking with @JohnTraverAmd and we were thinking there might be a case where the vendor signs with algorithm A but the owner wants to sign with algorithm B.

I think we should try to come up with a way to support this. Concretely, I could see an owner and vendor disagreeing on which algorithm provides better security properties between LMS and ML-DSA.

This doesn't need to block this PR, we can address it in a follow-up. But I just wanted to see if you have any thoughts on this.

Reply:
The public key format is generic enough to support this.

The biggest problem is that each combination will have a different-sized manifest, so it would be more challenging to write code to support reading the manifest (might not be able to just use a struct and zerocopy).

Reply:
Let's discuss this on the Thursday sync.